src/vs/workbench/contrib/notebook/browser/view/cellParts/cellComments.ts

```diff
@@ -87,41 +82,46 @@ export class CellComments extends CellContentPart {
 		await this._commentThreadWidget.display(layoutInfo.fontInfo.lineHeight, true);
 		this._applyTheme();
 
-		this.commentTheadDisposables.add(this._commentThreadWidget.onDidResize(() => {
+		this._commentThreadDisposables.add(this._commentThreadWidget.onDidResize(() => {
 			if (this.currentElement?.cellKind === CellKind.Code && this._commentThreadWidget) {
 				this.currentElement.commentHeight = this._calculateCommentThreadHeight(this._commentThreadWidget.getDimensions().height);
 			}
 		}));
 	}
 
 	private _bindListeners() {
-		this.cellDisposables.add(this.commentService.onDidUpdateCommentThreads(async () => {
-			if (this.currentElement) {
-				const info = await this._getCommentThreadForCell(this.currentElement);
-				if (!this._commentThreadWidget && info) {
-					await this._createCommentTheadWidget(info.owner, info.thread);
-					const layoutInfo = (this.currentElement as CodeCellViewModel).layoutInfo;
-					this.container.style.top = `${layoutInfo.outputContainerOffset + layoutInfo.outputTotalHeight}px`;
-					this.currentElement.commentHeight = this._calculateCommentThreadHeight(this._commentThreadWidget!.getDimensions().height);
-					return;
-				}
-
-				if (this._commentThreadWidget) {
-					if (!info) {
-						this._commentThreadWidget.dispose();
-						this.currentElement.commentHeight = 0;
-						return;
-					}
-					if (this._commentThreadWidget.commentThread === info.thread) {
-						this.currentElement.commentHeight = this._calculateCommentThreadHeight(this._commentThreadWidget.getDimensions().height);
-						return;
-					}
-
-					await this._commentThreadWidget.updateCommentThread(info.thread);
-					this.currentElement.commentHeight = this._calculateCommentThreadHeight(this._commentThreadWidget.getDimensions().height);
-				}
+		this.cellDisposables.add(this.commentService.onDidUpdateCommentThreads(async () => this._updateThread()));
+	}
+
+	private async _updateThread() {
+		if (!this.currentElement) {
+			return;
+		}
+		const info = await this._getCommentThreadForCell(this.currentElement);
+		if (!this._commentThreadWidget && info) {
+			await this._createCommentTheadWidget(info.owner, info.thread);
+			const layoutInfo = (this.currentElement as CodeCellViewModel).layoutInfo;
+			this.container.style.top = `${layoutInfo.outputContainerOffset + layoutInfo.outputTotalHeight}px`;
+			this.currentElement.commentHeight = this._calculateCommentThreadHeight(this._commentThreadWidget!.getDimensions().height);
+			return;
+		}
+
+		if (this._commentThreadWidget) {
+			if (!info) {
+				this._commentThreadDisposables.clear();
+				this._commentThreadWidget.dispose();
+				this._commentThreadWidget = null;
```

Comment:
@alexr00 a MutableDisposable is a nicer pattern for as it helps you not forget to dispose when unsetting

Reply:
Thanks, I have changed the code to use this.